### PR TITLE
Add check for adding gpu tag suffix when image field is specified

### LINKED ISSUE #2708: There is an additional word after pytorch/torchserve-kfs:0.6.0-gpu
/kind bug

What steps did you take and what happened:
[A clear and concise description of what the bug is.]

kserve output 'pytorch/torchserve-kfs:0.6.0-gpu-gpu' instead of 'pytorch/torchserve-kfs:0.6.0-gpu', after I rollback to kserve0.9.0, it recovers.

source yaml file 
```yaml
apiVersion: serving.kserve.io/v1beta1
kind: InferenceService
metadata:
  name: intention-and-similar
spec:
  predictor:
    model:
      image: pytorch/torchserve-kfs:0.6.0-gpu
      modelFormat:
        name: pytorch
      name: ''
      resources:
        limits:
          memory: 10Gi
          nvidia.com/gpu: '1'
        requests:
          memory: 9Gi
          nvidia.com/gpu: '1'
      runtime: kserve-torchserve
      storageUri: pvc://ai-pv-claim/intention-and-similar
```

log in kserve-controller-manager-79ccd97ff-rtjhn pod:
```log
{"level":"info","ts":1676883977.3338618,"logger":"PredictorReconciler","msg":"Resolved container","container":"&Container{Name:kserve-container,Image:pytorch/torchserve-kfs:0.6.0-gpu-gpu,Command:[],Args:[torchserve --start --model-store=/mnt/models/model-store --ts-config=/mnt/models/config/config.properties],WorkingDir:,Ports:[]ContainerPort{},Env:[]EnvVar{EnvVar{Name:TS_SERVICE_ENVELOPE,Value:kserve,ValueFrom:nil,},},Resources:ResourceRequirements{Limits:ResourceList{cpu: {{1 0} {<nil>} 1 DecimalSI},memory: {{10737418240 0} {<nil>} 10Gi BinarySI},nvidia.com/gpu: {{1 0} {<nil>} 1 DecimalSI},},Requests:ResourceList{cpu: {{1 0} {<nil>} 1 DecimalSI},memory: {{9663676416 0} {<nil>} 9Gi BinarySI},nvidia.com/gpu: {{1 0} {<nil>} 1 DecimalSI},},},VolumeMounts:[]VolumeMount{},LivenessProbe:nil,ReadinessProbe:nil,Lifecycle:nil,TerminationMessagePath:,ImagePullPolicy:,SecurityContext:nil,Stdin:false,StdinOnce:false,TTY:false,EnvFrom:[]EnvFromSource{},TerminationMessagePolicy:,VolumeDevices:[]VolumeDevice{},StartupProbe:nil,}","podSpec":{"containers":[{"name":"kserve-container","image":"pytorch/torchserve-kfs:0.6.0-gpu-gpu","args":["torchserve","--start","--model-store=/mnt/models/model-store","--ts-config=/mnt/models/config/config.properties"],"env":[{"name":"TS_SERVICE_ENVELOPE","value":"kserve"}],"resources":{"limits":{"cpu":"1","memory":"10Gi","nvidia.com/gpu":"1"},"requests":{"cpu":"1","memory":"9Gi","nvidia.com/gpu":"1"}}}],"serviceAccountName":"xiaowei-ai"}}
{"level":"info","ts":1676883977.3429856,"logger":"DeploymentReconciler","msg":"Deployment Updated","Diff":"  v1.DeploymentSpec{\n  \t... // 1 ignored field\n  \tSelector: &{MatchLabels: {\"app\": \"isvc.intention-and-similar-predictor-default\"}},\n  \tTemplate: v1.PodTemplateSpec{\n  \t\tObjectMeta: {Name: \"intention-and-similar-predictor-default\", Namespace: \"xiaowei-dev-default\", Labels: {\"app\": \"isvc.intention-and-similar-predictor-default\", \"component\": \"predictor\", \"serviceEnvelope\": \"kserve\", \"serving.kserve.io/inferenceservice\": \"intention-and-similar\"}, Annotations: {\"internal.serving.kserve.io/storage-initializer-sourceuri\": \"s3://xiaowei-ai-models/intention-and-similar_2022-11-03/\", \"kustomize.toolkit.fluxcd.io/reconcile\": \"disabled\", \"serving.kserve.io/deploymentMode\": \"RawDeployment\"}, ...},\n  \t\tSpec: v1.PodSpec{\n  \t\t\tVolumes:        nil,\n  \t\t\tInitContainers: nil,\n  \t\t\tContainers: []v1.Container{\n  \t\t\t\t{\n  \t\t\t\t\tName: \"kserve-container\",\n  \t\t\t\t\tImage: strings.Join({\n  \t\t\t\t\t\t\"pytorch/torchserve-kfs:0.6.0-gpu\",\n- \t\t\t\t\t\t\"-gpu\",\n  \t\t\t\t\t}, \"\"),\n  \t\t\t\t\tCommand: nil,\n  \t\t\t\t\tArgs:    {\"torchserve\", \"--start\", \"--model-store=/mnt/models/model-store\", \"--ts-config=/mnt/models/config/config.properties\"},\n  \t\t\t\t\t... // 18 identical fields\n  \t\t\t\t},\n  \t\t\t},\n  \t\t\tEphemeralContainers: nil,\n  \t\t\tRestartPolicy:       \"Always\",\n  \t\t\t... // 31 identical fields\n  \t\t},\n  \t},\n  \tStrategy:        {Type: \"RollingUpdate\", RollingUpdate: &{MaxUnavailable: &{Type: 1, StrVal: \"25%\"}, MaxSurge: &{Type: 1, StrVal: \"25%\"}}},\n  \tMinReadySeconds: 0,\n  \t... // 3 identical fields\n  }\n"}
{"l
```
 
What did you expect to happen:

What's the InferenceService yaml:
[To help us debug please run kubectl get isvc $name -n $namespace -oyaml and paste the output]

Anything else you would like to add:
[Miscellaneous information that will assist in solving the issue.]

Environment:

Istio Version: 1.13.5
Knative Version: None
KServe Version: v0.10.0
Kubeflow version:
Cloud Environment:[k8s_istio/istio_dex/gcp_basic_auth/gcp_iap/aws/aws_cognito/ibm]
Minikube/Kind version:
Kubernetes version: (use kubectl version): v1.22.15-eks-fb459a0
OS (e.g. from /etc/os-release):
NAME="Amazon Linux"
VERSION="2"
ID="amzn"
ID_LIKE="centos rhel fedora"
VERSION_ID="2"
PRETTY_NAME="Amazon Linux 2"

### PR DESCRIPTION
**Pull request by yuzisun**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2708 

**Type of changes**
Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevent result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/controller/v1beta1/inferenceservice/utils/utils.go [hunk 1]
```diff
@@ -258,7 +258,10 @@ func UpdateImageTag(container *v1.Container, runtimeVersion *string, servingRunt
 				// For TFServing/TorchServe the GPU image is tagged with suffix "-gpu", when the version is found in the tag
 				// and runtimeVersion is not specified, we default to append the "-gpu" suffix to the image tag
 				if servingRuntime != nil && (*servingRuntime == constants.TFServing || *servingRuntime == constants.TorchServe) {
-					container.Image = image + "-gpu"
+					//check for the case when image field is specified directly with gpu tag
+					if !strings.HasSuffix(container.Image, "-gpu") {
+						container.Image = image + "-gpu"
```
Comment (rachitchauhan43) [inline pkg/controller/v1beta1/inferenceservice/utils/utils.go:263]: Why do we need to add `-gpu` tag and not just directly pick the image field as it is ? 
Like this:
`container.Image = image`

Comment (yuzisun) [reply pkg/controller/v1beta1/inferenceservice/utils/utils.go:263]: The torchserve cpu and gpu image has different tags, on the serving runtime it defines the cpu image tag so we need to append the gpu suffix based on the resource requirement user defined on inference service yaml.

Comment (rachitchauhan43) [reply pkg/controller/v1beta1/inferenceservice/utils/utils.go:263]: Got it.

**Review by Suresh-Nakkeran — APPROVED** [review]

**Comment by Suresh-Nakkeran** [comment]
\lgtm

**Pull request merged**